### PR TITLE
M3-5759: Don't make /account requests if restricted user

### PR DESCRIPTION
**Pull request by HanaXu**
## Description
Once Cloud has established that a user is restricted, we should not attempt to request any of the account endpoints

## How to test
Go to `/account` on a restricted user and click through the tabs while verifying that we longer make requests to /account endpoints

### CODE REVIEW

**Review by bnussman — CHANGES_REQUESTED** [review]
Works great and implementation is super simple! 🚀 I found one issue that might apply to some of the queries.

File: packages/manager/src/queries/account.ts [hunk 1]
```diff
@@ -5,16 +5,21 @@ import {
...
 
-export const useAccount = () =>
-  useQuery<Account, APIError[]>(queryKey, getAccountInfo, {
+export const useAccount = () => {
+  const { data: profile } = useProfile();
+
+  return useQuery<Account, APIError[]>(queryKey, getAccountInfo, {
     ...queryPresets.oneTimeFetch,
     ...queryPresets.noRetry,
+    enabled: !profile?.restricted,
```
Comment (johnwcallahan) [inline packages/manager/src/queries/account.ts:20]: Is the non-null assertion operator necessary here? (and similar places)

Comment (codyfinn) [reply packages/manager/src/queries/account.ts:20]: Do you mean the optional chaining operator?

File: packages/manager/src/queries/account.ts [hunk 2]
```diff
@@ -5,16 +5,21 @@ import {
...
+import { useProfile } from 'src/queries/profile';
 import { getGravatarUrl } from 'src/utilities/gravatar';
 import { mutationHandlers, queryPresets } from './base';
 
 export const queryKey = 'account';
 
-export const useAccount = () =>
-  useQuery<Account, APIError[]>(queryKey, getAccountInfo, {
+export const useAccount = () => {
+  const { data: profile } = useProfile();
```
[the remark this answers is not in the document]
Comment (bnussman) [reply packages/manager/src/queries/account.ts:15]: The profile is fetched and cached whenever Cloud Manager loads (see `AuthenticationWrapper.tsx`). So making queries dependent on the `useProfile` query should be fine. I won't actually make extra https requests unless somehow the profile is evicted from the cache.

File: packages/manager/src/queries/account.ts (hunk 1 above)
Comment (johnwcallahan) [reply packages/manager/src/queries/account.ts:20]: Yeah, I was confusing the leading `!` with a non-null assertion; that's my mistake. Thanks for the discussion @codyfinn and @bnussman.

File: packages/manager/src/queries/accountPayment.ts [hunk 3]
```diff
@@ -18,11 +23,14 @@ export const usePaymentMethodsQuery = (params: any = {}, filter: any = {}) => {
...
 
 export const useAllPaymentMethodsQuery = () => {
+  const { data: grants } = useGrants();
+
   return useQuery<PaymentMethod[], APIError[]>(
     queryKey + '-all',
     getAllPaymentMethodsRequest,
     {
       ...queryPresets.oneTimeFetch,
+      enabled: grants?.global.account_access !== null,
```
Comment (bnussman) [inline packages/manager/src/queries/accountPayment.ts:33]: Screen Shot 2022-07-01 at 12 40 31 PM

The `!== null` seems spot on. I think we just need this optional chain to not crash non-restricted accounts

```suggestion
      enabled: grants?.global?.account_access !== null,
```

**Review by bnussman — CHANGES_REQUESTED** [review]

**Review by codyfinn — APPROVED** [review]
Looks good to me! Might want to add tests but I don't think that is blocking.

**Review by bnussman — APPROVED** [review]
Nice work on this one 🚀

**Pull request merged**